feat: add `stats/incr/nanmsumprod`

Girish-Garg opened the pull request:
Resolves #5613 .

## Description

> What is the purpose of this pull request?

This pull request:

-   feat: add incrnanmsumprod accumulator for computing moving sum of products while ignoring NaN values by making a wrapper over msumprod function which pre-exisits at `stats/incr`

test:
![image](https://github.com/user-attachments/assets/897502f5-1ae4-452c-b2cb-62dd2abae40f)

examples:
![image](https://github.com/user-attachments/assets/b207b0db-aa21-468d-9c78-dfd94d80ce62)

benchmark:
![image](https://github.com/user-attachments/assets/543cdbbb-f64a-4b3c-ba68-e61fa9c6e786)

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5613  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Girish-Garg committed:
feat: add incrnanmsumprod accumulator for computing moving sum of products while ignoring NaN values

Girish-Garg commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented:
@Girish-Garg You'll want to resolve the CI failures.

Girish-Garg committed:
Merge branch 'stdlib-js:develop' into feature/nanmsumprod

Girish-Garg committed:
chore: fix linting and clean-up

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmsumprod/test/test.js:
```diff
@@ -0,0 +1,147 @@
+// MODULES //
+
+var tape = require( 'tape' );
+var incrnanmsumprod = require( '@stdlib/stats/incr/nanmsumprod/lib' );
```
Style: Test files in stdlib should use relative paths to require the local package, not absolute package paths.

```suggestion
var incrnanmsumprod = require( './../lib' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmsumprod/test/test.js:
```diff
@@ -0,0 +1,147 @@
+		[ 2.0, NaN ],
+		[ 3.0, -5.0 ],
+		[ NaN, 3.0 ],
+		[ NaN, NaN],
```
Style: Missing space before closing bracket to match stdlib spacing conventions.

```suggestion
		[ NaN, NaN ]
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmsumprod/lib/main.js:
```diff
@@ -0,0 +1,91 @@
+
+// MODULES //
+
+var incrmsum = require( '@stdlib/stats/incr/msumprod' );
```
Misleading variable name. This imports `msumprod`, not `msum`.

```suggestion
var incrmsumprod = require( '@stdlib/stats/incr/msumprod' );
```

Also update the usage on line 69 accordingly.

Planeshifter committed:
chore: address review feedback and update copyright years

- use relative require path in test file
- fix bracket spacing in test fixture
- use `format` for benchmark name string concatenation

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/nanmsumprod

Planeshifter committed:
chore: address review feedback

- rename require variable `incrmsum` to `incrmsumprod` to match wrapped package
- remove unused `isnan` import from README example
- clear hand-populated See Also/related-links sections for auto-population
